### PR TITLE
Concurrency

### PR DESCRIPTION
**Pull request by nacho692**
This PR allows concurrent access to DataSourceWithCache. Note that the comitted caché must be synchronized both for reads and writes due to having an LRU LinkedHashMap, which is not thread safe on reads.
*fed:version120* *fit:120*

### CODE REVIEW

File: rskj-core/src/main/java/org/ethereum/datasource/DataSourceWithCache.java [hunk 1]
```diff
@@ -123,20 +150,30 @@ public void updateBatch(Map<ByteArrayWrapper, byte[]> rows, Set<ByteArrayWrapper
...
 
     @Override
     public synchronized void flush() {
-        Map<ByteArrayWrapper, byte[]> uncommittedBatch = uncommittedCache.entrySet().stream().filter(e -> e.getValue() != null).collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
-        Set<ByteArrayWrapper> uncommittedKeysToRemove = uncommittedCache.entrySet().stream().filter(e -> e.getValue() == null).map(Map.Entry::getKey).collect(Collectors.toSet());
-        base.updateBatch(uncommittedBatch, uncommittedKeysToRemove);
-        committedCache.putAll(uncommittedCache);
-        uncommittedCache.clear();
+        try {
+            lock.writeLock().lock();
```
Comment (lsebrie) [inline rskj-core/src/main/java/org/ethereum/datasource/DataSourceWithCache.java]: Move this line out of `try {` statement

**Comment by lsebrie** [comment]
Pipeline: run

**Comment by lsebrie** [comment]
Pipeline: run

**Comment by lsebrie** [comment]
Pipeline: run

**Comment by nacho692** [comment]
Pipeline: run

**Pull request merged**